mkosi/__init__.py

```diff
@@ -3203,11 +3204,26 @@ def nspawn_params_for_build_sources(args: MkosiArgs, sft: SourceFileTransfer) ->
     else:
         params += ["--chdir=/root"]
 
-    params += [f"--setenv={env}={value}" for env, value in args.environment.items()]
+    params += [
+        f"--setenv={env}={value}"
+        for env, value in extend_run_env(args, args.environment).items()
+    ]
```

Comment:
Looking at the call sites of `nspaw_params_for_build_sources`, I see that the variables are already set for build scripts, so those lines should be removed then.

Reply:
When I look the call sites are [here](https://github.com/systemd/mkosi/blob/59aa8283d5ce564d5ea7c04103168e4cf35eb552/mkosi/__init__.py#L3179) in `run_prepare_script()` and [here](https://github.com/systemd/mkosi/blob/59aa8283d5ce564d5ea7c04103168e4cf35eb552/mkosi/__init__.py#L7394) in `run_build_script()`. It's the second call site that sets these variables but not the first. I'll remove [these lines](https://github.com/systemd/mkosi/blob/59aa8283d5ce564d5ea7c04103168e4cf35eb552/mkosi/__init__.py#L7388-#L7392) from `run_build_script()` to compensate.